### ISSUE TITLE
Question: ParallelStreamProcessor::pollAndProduce with different key or value to produce

### ISSUE THREAD

**Issue by acktsap** [confluentinc/parallel-consumer#502]
ParallelStreamProcessor::pollAndProduce only supports producing with the same key-value with the consumer.

https://github.com/confluentinc/parallel-consumer/blob/78aab492941dc647ea4a828bb4723305267157cd/parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelStreamProcessor.java#L59-L65

I want to produce message with different key, value format. Since `ParallelStreamProcessor::pollAndProduce` only supports producing with the same key-value, i used `ParallelStreamProcessor::poll` and separate producer.

```java
Consumer<String, String> kafkaConsumer = new KafkaConsumer<>(consumerProperties);
var options = ParallelConsumerOptions.<String, String>builder()
    .ordering(ParallelConsumerOptions.ProcessingOrder.KEY)
    .maxConcurrency(1000)
    .consumer(kafkaConsumer)
    .build();
ParallelStreamProcessor<String, String> parallelStreamProcessor = ParallelStreamProcessor.createEosStreamProcessor(options);
parallelStreamProcessor.subscribe(List.of(TOPIC_NAME));

Producer<String, Integer> kafkaProducer = new KafkaProducer<>(producerProperties);

parallelStreamProcessor.poll(pollContext -> {
    logger.info("consumes: {}", pollContext);
    ProducerRecord<String, Integer> producerRecord = new ProducerRecord<>("newTopic", 3);
    logger.info("produces: {}", producerRecord);
    kafkaProducer.send(producerRecord); // uses separate producer
});
```

It there any way to use pollAndProduce in this case? Maybe code snippet would be

```java
Consumer<String, String> kafkaConsumer = new KafkaConsumer<>(consumerProperties);
Producer<String, Integer> kafkaProducer = new KafkaProducer<>(producerProperties);
var options = ParallelConsumerOptions.<String, String, String, Integer>builder() // need more generic
    .ordering(ParallelConsumerOptions.ProcessingOrder.KEY)
    .maxConcurrency(1000)
    .consumer(kafkaConsumer)
    .producer(kafkaProducer)
    .build();
ParallelStreamProcessor<String, String> parallelStreamProcessor = ParallelStreamProcessor.createEosStreamProcessor(options);
parallelStreamProcessor.subscribe(List.of(TOPIC_NAME));

parallelStreamProcessor.pollAndProduce(pollContext -> {
    logger.info("consumes: {}", pollContext);
    ProducerRecord<String, Integer> producerRecord = new ProducerRecord<>("newTopic", 3);
    logger.info("produces: {}", producerRecord);
    return producerRecord; // just returns new record
});
```

**Comment by astubbs** [comment]
Hi, not at the moment. See:
- #175 

It is intended to be fixed for 1.0 though. 

Workaround, is to tell PC to do byte,byte, and do your own serve in your function. So, given the workaround isn't a big deal, it's not been a priority, and I want to fix it right.

**Comment by acktsap** [comment]
Thank for reply and workaround! 

I didn't know there was an issue already. Close this one.
